src/plot_api/edit_types.js

```diff
@@ -54,5 +58,53 @@ module.exports = {
                 flags[editTypeParts[i]] = true;
             }
         }
-    }
+    },
+
+    overrideAll: overrideAll
 };
+
+/*
+ * For attributes that are largely copied from elsewhere into a plot type that doesn't
+ * support partial redraws - overrides the editType field of all attributes in the object
+ *
+ * @param {object} attrs: the attributes to override. Will not be mutated.
+ * @param {string} editTypeOverride: the new editType to use
+ * @param {bool|'nested'} overrideContainers: should we override editType for containers or just
+ *   `valObject`s? 'nested' will override editType for all but the top level. Containers below
+ *   the absolute top level (trace or layout root) DO need an editType, to handle the case
+ *   where you edit the whole container, but in some cases you want to provide it explicitly
+ *   as it may be more expansive than the attribute editTypes, since you don't know if there
+ *   is a data array in the container (which if provided directly triggers an automatic docalc)
+ */
+function overrideAll(attrs, editTypeOverride, overrideContainers) {
```

Comment:
I'd vote for a making the call signature `overrideAll(attrs, editTypeOverride, opts)` where `opts` can have key `overrideContainers`. It would make lines like this [one](https://github.com/plotly/plotly.js/pull/1999/commits/f49ae5ea580d5826666574c041a744e89180d873#diff-8e9a79b541026f6567178f4c9d990c99R288) more readable.

Reply:
Actually, `overrideAll(attrs, editTypeOverride, 'nested')` is very readable, so maybe that third argument could accept values `'nested'` and 'from-root'` (instead of `true`).

Reply:
Turns out we weren't using the default no-containers behavior at all - I guess if that's the behavior you needed, some of the other attributes would have different editTypes already so you couldn't use `overrideAll` at all - so I changed it to only `'nested'` and `'from-root'` - https://github.com/plotly/plotly.js/pull/1999/commits/b9826c81c0b5ab49048e0e601825aa045402aa5e